_posts/2018-11-13-deploy-rails.md

```diff
@@ -0,0 +1,146 @@
+
+#### Capistrano
+
+> A remote server automation and deployment tool written in Ruby.
+
+Capistrano allows us to write *tasks* that define our entire(almost) deployment workflow and then perform those tasks on any number of servers.
+Cap builds upon [Rake](https://github.com/ruby/rake/blob/master/doc/rational.rdoc) "Ruby Make", adding the functionality necessary to connect to and run code on remote servers.
+The cap [SSHKit](https://github.com/capistrano/sshkit) toolkit provides most of capistrano's functionality, and has useful [examples](https://github.com/capistrano/sshkit/blob/master/EXAMPLES.md).
+
+Various gems are available to integrate capistrano with rails applications:
+- capistrano/rails - common rails deployment tasks
+- capistrano/rvm - integration with RVM managed ruby environments
+- capistrano/bundler - integration with bundler
+- capistrano/puma - tasks to manage the puma application server
+
+With cap [added to a rails application](https://www.digitalocean.com/community/tutorials/deploying-a-rails-app-on-ubuntu-14-04-with-capistrano-nginx-and-puma#step-6-%E2%80%94-adding-deployment-configurations-in-the-rails-app), a `cap [environment] deploy` is all we need to run to deploy a new version of our site.
+Capistrano also provides the ability to rollback a deploy if the most recent one breaks.
+That by itself is reason enough to *capify* all the things.
+
+Since cap connects to the server via SSH, we can use it to run any command as we normally would on the server.
+By writing `tasks`, scripts that list the commands to run, both locally and on remote servers, we can automate the process of setting up our server.
+One only needs set up the server once, while extracting the logic out into cap tasks.
+In this way, we can have a tasks to install postgres, another for nginx, one to create a database role, one to update the system, and so on.
+Once you have the scripts together, provisioning a new server stops being a pain. But how do you get those tasks into a different project?
+[Rails application templates](https://guides.rubyonrails.org/rails_application_templates.html) is how.
+
+Rails application templates provide a way to add 'code' to a new or existing rails application.
+The dsl provides actions to add gems and sources to the `Gemfile`, add initializers, rake tasks and arbitrary files.
+We can set rails configuration variables or even run rails and git commands as the template is applied.
+We can use [Thor actions](https://www.rubydoc.info/github/wycats/thor/Thor/Actions#source_paths-instance_method) to fetch and manipulate data from disparate sources for use in our cap tasks.
+To generate a new application and apply the template:
+~~~shell
+rails new [appname] -m /path/to/template.rb # path can also be a url
+~~~
+Or to apply to an existing app:
+~~~shell
+bin/rails app:template LOCATION=/path/to/template.rb # location variable can also be a url
+~~~
+
+#### Example you ask?
+
```

Comment:
add a relevant image to draw attention. also include the first steps that help people to get going with rails_seed.

Reply:
Mention that rails seed is an app template that we developed at zegetech